CRM/Contribute/Form/Contribution/Confirm.php

```diff
@@ -1780,6 +1785,14 @@ protected function processSecondaryFinancialTransaction($contactID, &$form, $tem
     );
 
     $result = array();
+
+    if ($tempParams['skipLineItem']) {
+      // We are not processing the line item here because we are processing a membership.
+      // So unset the total_amount param, which stores the non-membership contribution amount.
+      $tempParams['total_amount'] = $membershipContribution->total_amount;
+      $tempParams['contributionID'] = $membershipContribution->id;
+    }
```

Comment:
I've tried to be more thorough about those parameters, though this is admittedly a "white-listing" approach which might be brittle. Is there a better way?